src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -380,8 +373,9 @@ class RedownloadService[F[_]: NonEmptyParallel](
 
       maxMajorityHeight = maxHeight(majorityState)
       ignorePoint = getIgnorePoint(maxMajorityHeight)
+      majorityStateCutoff = if (isDownload) maxMajorityHeight else ownPeer.getOrElse(maxMajorityHeight)
       meaningfulAcceptedSnapshots = takeHighestUntilKey(acceptedSnapshots, ignorePoint)
-      meaningfulMajorityState = takeHighestUntilKey(majorityState, ignorePoint)
+      meaningfulMajorityState = majorityState.filterKeys(_ >= majorityStateCutoff)
```

Comment:
why you removed `takeHighestUntilKey`?

Reply:
Probably because it uses `>` instead of `>=`

Reply:
takeHighest would result in `filterKeys(_ > majorityStateCutoff)` (greater than, instead of greater than equal)

Reply:
but it does not take into account the ignore point, which has been used before, does it?

Reply:
No, it doesn't. The ignore point is just `maxMajorityHeight - 40`.

Reply:
So it's wrong IMO. It should ignore everything below joining height but from the `takeHighestUntilKey(majorityState, ignorePoint)`, so `takeHighestUntilKey(majorityState, ignorePoint).filterKeys(_ >= majorityStateCutoff)` not from `majorityState` directly

Reply:
So it should ignore everything below both ignorePoint and majorityStateCutoff, whatever is higher

Reply:
@marcinwadon Do you think that should I remove the `ignorePoint`? The `ignorePoint` is used to trim down the `meaningfulAcceptedSnapshots`.

Reply:
Why remove? I don't understand

Reply:
`takeHighestUntilKey(majorityState, ignorePoint).filterKeys(_ >= majorityStateCutoff)` that will be no different than current implementation.

Reply:
tl;dr; there is a case that joining height can be lower than the ignore point and then the ignore point should be taken into account

Reply:
Resolved. @marcinwadon was right, the `ignorePoint` filter has to stay.

Reply:
`majorityStateCutoff` is I guess always above `ignorePoint` so

`majorityState.filterKeys(_ >= majorityStateCutoff)` and `takeHighestUntilKey(majorityState, ignorePoint).filterKeys(_ >= majorityStateCutoff)` produce the same @marcinwadon

Reply:
@kpudlik no. 

let's say max majorityState is 100
own joining height is 40

`majorityState.filterKeys(_ >= majorityStateCutoff)` will return `40+`
but it should produce `60+` (100 - majority state inverval (40))

Reply:
Ah right, I see that it is applied to both redownload and download. Then it is wrong implementation.